File: test/electron/rag/chunker-fixtures/sample.js
// JS fixture — uses CommonJS + function expressions.
const fs = require('fs');

function greet(name) {
  return `hello ${name}`;
}

class Greeter {
  constructor(prefix) {
    this.prefix = prefix;
  }
  greet(name) {
    return `${this.prefix} ${name}`;
  }
}

module.exports = { greet, Greeter };

File: crates/rag/test-fixtures/chunker/sample.py
# Python fixture — decorated defs, class, module constant.
import os

MAX_RETRIES = 3


def greet(name: str) -> str:
    return f"hello {name}"


class Greeter:
    def __init__(self, prefix):
        self.prefix = prefix

    def greet(self, name):
        return f"{self.prefix} {name}"


@property
def config(self):
    return self._config
